plugins/builds/index.js

```diff
@@ -221,33 +221,35 @@ exports.register = (server, options, next) => {
         };
 
         return pipelineFactory.get(pipelineId)
-            .then((pipeline) => {
-                const scmUri = pipeline.scmUri;
-                const admin = Object.keys(pipeline.admins)[0];
-                const scmContext = pipeline.scmContext;
-
-                payload.scmContext = scmContext;
-                payload.username = admin;
-
-                // get pipeline admin's token
-                return userFactory.get({ username: admin, scmContext })
-                    .then(user => user.unsealToken())
-                    .then((token) => {
-                        const scmConfig = {
-                            scmContext,
-                            scmUri,
-                            token
-                        };
-
-                        // Get commit sha
-                        return scm.getCommitSha(scmConfig)
-                            .then((sha) => {
-                                payload.sha = sha;
-
-                                return eventFactory.create(payload);
-                            });
-                    });
-            });
+            .then(pipeline => pipeline.admin
+                .then(realAdmin => realAdmin.username)
+                .then((adminName) => {
+                    const scmUri = pipeline.scmUri;
+                    const admin = adminName;
+                    const scmContext = pipeline.scmContext;
+
+                    payload.scmContext = scmContext;
+                    payload.username = admin;
+
+                    // get pipeline admin's token
+                    return userFactory.get({ username: admin, scmContext })
```

Comment:
No need to do this since `pipeline.admin` is already an user object, should be able to do `readAdmin.unsealToken()` directly. https://github.com/screwdriver-cd/models/pull/247/files

Reply:
I fixed. Thank you!